src/dlfcn.c

```diff
@@ -474,7 +479,207 @@ char *dlerror( void )
     return error_buffer;
 }
 
-#ifdef SHARED
+/* see https://docs.microsoft.com/en-us/archive/msdn-magazine/2002/march/inside-windows-an-in-depth-look-into-the-win32-portable-executable-file-format-part-2
+ * for details */
+
+/*
+ * Return pointer to import directory
+ */
+static IMAGE_IMPORT_DESCRIPTOR* get_import_directory( HMODULE module, DWORD *size )
+{
+   IMAGE_DOS_HEADER *dosHeader;
+   IMAGE_OPTIONAL_HEADER *optionalHeader;
+
+   dosHeader = (IMAGE_DOS_HEADER *) module;
+   if( dosHeader->e_magic != 0x5A4D )
+       return NULL;
+   optionalHeader = (IMAGE_OPTIONAL_HEADER *) ((BYTE *) module + dosHeader->e_lfanew + 24);
+    if ( optionalHeader->Magic != IMAGE_NT_OPTIONAL_HDR_MAGIC )
+        return NULL;
+    if ( optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size == 0 ||
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress == 0 )
+        return NULL;
+    *size = optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].Size;
+    return (IMAGE_IMPORT_DESCRIPTOR*)((BYTE*)module +
+        optionalHeader->DataDirectory[IMAGE_DIRECTORY_ENTRY_IMPORT].VirtualAddress);
+}
+
+/*
+ * Return pointer to import address table
+ */
+static IMAGE_IMPORT_DESCRIPTOR* get_import_address_table( HMODULE module, DWORD *size )

```

Comment:
Looking at it again and now I see why I was confused. IAT is not `IMAGE_IMPORT_DESCRIPTOR` structure, but rather it is an array of 32 (or 64) bit integers. And therefore return type of this function is not correct. 

https://docs.microsoft.com/en-us/windows/win32/debug/pe-format?redirectedfrom=MSDN#import-address-table

I would suggest to use `void *` as return type of this function as `void *` is used also in get_address_from_import_address_table for iat parameter.